File: src/utils/ApiError.js
class ApiError extends Error {
  constructor(
    statusCode,
    message = "Something went wrong",
    errors = [],
    errorStackTrace = "",
  ) {
    super(message);
    this.message = message;
    this.statusCode = statusCode;
    this.data = null;
    this.errors = errors;
    this.success = false;

    if (errorStackTrace) {
      this.stack = errorStackTrace;
    } else {
      Error.captureStackTrace(this, this.constructor);
    }
  }
}

export { ApiError };

File: frontend/flutter/lib/common/widgets/api_error.dart
class ApiError extends Error {
  int statusCode;
  String message;
  List<dynamic> errors;
  dynamic data;
  bool success;

  ApiError({
    required this.statusCode,
    required this.message,
    this.data = null,
    this.errors = const [],
    this.success = false,
  });

  factory ApiError.fromMap(Map<String, dynamic> map) {
    return ApiError(
        statusCode: map['statusCode'],
        message: map['message'],
        data: map['data'],
        errors: map['errors'],
        success: map['success']);
  }
  Map<String, dynamic> toMap() {
    return {
      "statusCode": statusCode,
      "message": message,
      "data": data,
      "errors": errors,
      "success": success,
    };
  }
}
